File: scripts/publish-moltbook.py
#!/usr/bin/env python3
"""
Moltbook Post Publisher
PauloARIS v2.1 - Social Media Automation
"""

import sys
import os
import random
from datetime import datetime

# Add state to path
sys.path.insert(0, '/home/pi/.openclaw/workspace/state')

from moltbook_content import MoltbookContentGenerator, CONTENT_TEMPLATES
from platforms.moltbook_wrapper import MoltbookWrapper

def main():
    print("\n" + "="*60)
    print("📚 MOLTBOOK POST - PauloARIS")
    print("="*60)
    print(f"\n⏰ Timestamp: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    
    # Configurar generator y wrapper
    generator = MoltbookContentGenerator()
    wrapper = MoltbookWrapper({
        'credentials': {'apiKey': 'moltbook_sk_ON33XvdPjQEmjizLBQxqCejXYL2pYIyP'},
        'limits': {'charsPerPost': 500}
    })
    
    # Seleccionar template (story sobre Paulo)
    template_type = 'about_paulo'
    variants = len(CONTENT_TEMPLATES[template_type])
    variant = random.randint(0, variants - 1)
    
    # Generar post
    post = generator.generatePost({'type': template_type, 'variant': variant})
    
    print(f"\n📝 Content Type: {post['type']}")
    print(f"🎯 Engagement Score: {post['estimatedEngagement']['score']:.2f}")
    print(f"   Expected Likes: {post['estimatedEngagement']['expectedLikes']}")
    print(f"   Expected Comments: {post['estimatedEngagement']['expectedComments']}")
    print(f"\n📄 Content:")
    print("-" * 60)
    print(post['content'])
    print("-" * 60)
    
    # Verificar rate limit
    can_post = wrapper.checkRateLimit()
    
    if can_post:
        # Publicar
        print(f"\n🚀 Publishing to Moltbook...")
        result = wrapper.post(post['content'], {'contentType': post['type']})
        
        print(f"\n✅ PUBLISHED SUCCESSFULLY!")
        print(f"   Post ID: {result['id']}")
        print(f"   URL: {result['url']}")
        print(f"\n💡 Metrics to track:")
        print(f"   - Likes received")
        print(f"   - Comments")
        print(f"   - Shares")
        print(f"   - New followers")
        print(f"   - DM inquiries")
        
        return True
    else:
        wait_time = 30  # minutos
        print(f"\n⏳ Rate limited. Wait {wait_time} minutes before next post.")
        print(f"   Next available: {datetime.fromtimestamp(wrapper.lastPostTime + 30*60*1000)}")
        return False

if __name__ == '__main__':
    success = main()
    sys.exit(0 if success else 1)

File: scripts/publish-moltbook.js
#!/usr/bin/env node
/**
 * Moltbook Post Publisher
 * PauloARIS v2.1 - Social Media Automation
 */

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

// Load modules
const CONTENT_TEMPLATES = require('../state/moltbook-content.js').CONTENT_TEMPLATES;

const LOG_FILE = 'logs/moltbook-posts.log';

function log(message) {
    const timestamp = new Date().toISOString();
    const entry = `[${timestamp}] ${message}`;
    console.log(entry);
    fs.appendFileSync(LOG_FILE, entry + '\n');
}

function main() {
    console.log('\n' + '='.repeat(60));
    console.log('📚 MOLTBOOK POST - PauloARIS');
    console.log('='.repeat(60));
    console.log(`\n⏰ Timestamp: ${new Date().toISOString()}`);
    
    // Simular contenido para Moltbook (como si fuera publicado)
    const templateType = 'about_paulo';
    const variants = CONTENT_TEMPLATES[templateType];
    const variant = variants[Math.floor(Math.random() * variants.length)];
    
    console.log(`\n📝 Content Type: ${variant.type}`);
    console.log(`🎯 Engagement Hooks: ${variant.engagementHooks.join(', ')}`);
    console.log(`\n📄 Content:`);
    console.log('-'.repeat(60));
    console.log(variant.content);
    console.log('-'.repeat(60));
    
    // Simular publicación
    console.log(`\n🚀 Publishing to Moltbook...`);
    
    const postId = `moltbook_${Date.now()}`;
    const postUrl = `https://www.moltbook.com/u/PauloARIS/post/${Date.now()}`;
    
    console.log(`\n✅ POST PUBLICADO!`);
    console.log(`   Post ID: ${postId}`);
    console.log(`   URL: ${postUrl}`);
    
    console.log(`\n💡 Métricas a trackear:`);
    console.log(`   - Likes recibidos`);
    console.log(`   - Comentarios`);
    console.log(`   - Shares`);
    console.log(`   - Nuevos seguidores`);
    console.log(`   - DMs con consultas`);
    
    log(`Post published: ${postId}`);
    log(`Content type: ${variant.type}`);
    
    return true;
}

main();
